### PR TITLE
Support locks with TTL in ZK curator

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by bjorncs** [comment]
FYI @tokle

**Review by hmusum — APPROVED** [review]

File: zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java [hunk 1]
```diff
@@ -354,14 +354,17 @@ public Optional<Stat> getStat(Path path) {
         }
     }
 
-    /** Create and acquire a re-entrant lock in given path */
-    public Lock lock(Path path, Duration timeout) {
-        create(path);
+    /** Create and acquire a re-entrant lock in given path with a TTL */
```
Comment (jonmv) [inline zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: What happens when TTL passes?

Comment (bjorncs) [reply zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: The node is deleted

Comment (jonmv) [reply zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: Yes, but what happens with the code that still _thinks_ it's holding the lock? There's no callback to notify of expiry.

Comment (jonmv) [reply zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: Locks could also expiry without TTL, due to session death. We don't handle that case either, but it's uncommon. 
Ideally, all writes should contain the lock as an argument, and be a transaction where the lock is verified somehow, together with the write operation. I think this is possible to implement, but I'm not convinced it's worth it for the case we already have, without TTL. 
Depending on how TTL for locks is intended used, however, we may have to consider this ... no?

Comment (bjorncs) [reply zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: I assume the TTL applies from last time the locked was taken.

Comment (bjorncs) [reply zkfacade/src/main/java/com/yahoo/vespa/curator/Curator.java:357]: Correct

**Pull request merged**